### ISSUE TITLE
Allow access to banned type members for instances of derived class

### ISSUE THREAD

**Issue by MaceWindu** [dotnet/roslyn-analyzers#5182]
### Analyzer

**Diagnostic ID**: [RS0030](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs)

### Describe the improvement

I have following scenario:
- there is an attribute that is not allowed to use directly (attribute type banned by rule)
- instead of this attribute we have new attribute type, derived from banned one

There are two issues with banned api analyzer I see:

1. Deriving from banned type doesn't rise `RS0030` (I would expect it to produce error that I need to supporess in this case)

```cs
// no RS0030 here
public class MyAttribute : BannedAttribute
{
}
```

2. access to banned type properties on derived type rise `RS0030`
```cs
// RS0030 generated here for Description property, defined in BannedAttribute
// but I didn't banned property explicitly and we deal with non-banned MyAttribute type here
[My(Description = "this is method")]
publict void Method() {}
```

Workaround:
```cs
public class MyAttribute : BannedAttribute
{
        public new string? Description
        {
#pragma warning disable RS0030 // Do not used banned APIs
            get => base.Description;
            set => base.Description = value;
#pragma warning restore RS0030 // Do not used banned APIs
        }
}
```
